frontend/lib/resolution/Resolver.cpp

```diff
@@ -2338,30 +2366,34 @@ void Resolver::exit(const TypeQuery* tq) {
 }
 
 bool Resolver::enter(const NamedDecl* decl) {
+
+  if (decl->id().postOrderId() < 0) {
+    // It's a symbol with a different path, e.g. a Function.
```

Comment:
is this specific to _nested_ functions or all functions? It might not matter much, just noticed the old comment called out nested functions.

Reply:
it applies to all functions but the code running Resolver visitors will visit the elements of the function individually as appropriate for that task (formals, body, etc). Of course we might change this in the future, but today, this check intentionally skips all Functions, because if we are resolving a Function, the visitor will see that as visiting formals and body etc individually.